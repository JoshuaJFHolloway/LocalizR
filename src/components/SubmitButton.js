import React from 'react';
import Button from './styledComponents/quizButtons.js';

const SubmitButton = props => {
  return (
      <Button id={"submitButton"} onClick={props.handleDataSubmitted} style={props.style}> {props.buttonName} </Button>
  );
};

export default SubmitButton;
