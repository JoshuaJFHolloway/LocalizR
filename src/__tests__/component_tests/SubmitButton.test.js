import React from 'react';
import { shallow } from 'enzyme';
import SubmitButton from '../../components/SubmitButton';

describe('Submit data', () => {
  let submit = shallow(<SubmitButton />);

  it('renders correctly', () => {
    expect(submit).toMatchSnapshot();
  });


  describe('when a user clicks the button', () => {

    it('should call the handleDataSubmitted function', () => {
      let mockhandleDataSubmitted = jest.fn();
      let submit = shallow(<SubmitButton handleDataSubmitted = {mockhandleDataSubmitted} />);
      submit.find("#submitButton").simulate('click');
      expect(mockhandleDataSubmitted).toBeCalled();
    })
  });

});