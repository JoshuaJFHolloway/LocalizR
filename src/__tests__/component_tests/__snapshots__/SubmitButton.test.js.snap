// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit data renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubmitButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": " Submit ",
      "id": "submitButton",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": " Submit ",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": " Submit ",
        "id": "submitButton",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": " Submit ",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
